src/Compiler/FSharp.Compiler.Service.fsproj

```diff
@@ -13,6 +13,15 @@
     <AssemblyName>FSharp.Compiler.Service</AssemblyName>
     <AllowCrossTargeting>true</AllowCrossTargeting>
     <DefineConstants>$(DefineConstants);COMPILER</DefineConstants>
+    <CheckNulls>true</CheckNulls>
+	<!-- Nullness checking againts ns20 base class libraries is very weak, the APIs were not updated with annotations. 
+	     Therefore we want to use the latest BCL APIs from NetCurrent.
+		 We are still not building the actual product in NetCurrent, so for official builds we remain as ns2.0.
+		 For 'BUILDING_USING_DOTNET' builds, we still want latest BCL annotations, so that contributors can get related warnings locally.
+
+     On CI, OSX has problems with Xliff targets for net9, skipping via SKIP_NETCURRENT_FSC_BUILD until resolved ( The target "UpdateXlf" does not exist in the project.)
+	-->
+    <TargetFrameworks Condition=" '$(OfficialBuildId)' == '' AND '$(FSharpNetCoreProductDefaultTargetFramework)' != '' AND '$(Configuration)' != 'Proto' AND '$(SKIP_NETCURRENT_FSC_BUILD)' != 'true' ">$(FSharpNetCoreProductDefaultTargetFramework);$(TargetFrameworks)</TargetFrameworks>
```

Comment:
One downside I can see (potentially) happening is:
1. Contributor loads up solution
2. net9.0 (or whatever the current one at the time) gets selected
3. Contributor adds a functionality which is only present in net9.0 BCL
4. Contributor runs tests for net9.0 (which results the rebuild of only net9.0), everything pass.
5. (Optionally) contributor updates/creates baselines for net9.0 unknowingly.
6. On PR/Push there are a bunch of confusing issues in CI, because netstandard is used as primary runtime libraries.

Reply:
This isn't replacing the TFM, it's multi-targeting - the old NS20 build is still happening.
Also, I changed test projects to explicitely load the NS20 version of F.C.S (when I did not, it did pick the net90 version, you are right).

Reply:
This is what was added on the F.C.S.Tests side:

```xml
<ProjectReference Include="..\..\src\Compiler\FSharp.Compiler.Service.fsproj">
		<!-- If tests would use net9.0, the surface area changes because of different set of Assembly references -->
		<SetTargetFramework>TargetFramework=netstandard2.0</SetTargetFramework>
	</ProjectReference>
```